vllm/config/model_arch.py

```diff
@@ -0,0 +1,145 @@
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Copyright contributors to the vLLM project
+import functools
+import json
+from dataclasses import field
+from typing import Any
+
+from pydantic import ConfigDict
+from pydantic.dataclasses import dataclass
+from torch import nn
+
+from vllm.config.utils import config
+from vllm.logger import init_logger
+
+logger = init_logger(__name__)
+
+
+@dataclass(config=ConfigDict(arbitrary_types_allowed=True, extra="allow"))
+class ModelArchitectureTextConfig:
+    model_type: str
+    hidden_size: int
+    num_hidden_layers: int
+    num_attention_heads: int
+    use_deepseek_mla: bool
+    head_dim: int
+    vocab_size: int
+    num_key_value_heads: int
+    num_experts: int
+
+    def __init__(
+        self,
+        model_type: str,
+        hidden_size: int,
+        num_hidden_layers: int,
+        num_attention_heads: int,
+        use_deepseek_mla: bool,
+        head_dim: int,
+        vocab_size: int,
+        num_key_value_heads: int,
+        num_experts: int,
+        **kwargs,
```

Comment:
Why do we need kwargs? Should ModelArchitectureTextConfig be a precise set of configs that vllm need?

Reply:
I thought our plan was to use a more standarized config to replace hf_config? If so, we will need the extra fields for model arch initialization.

I'm also okay with keeping both hf_config and model_arch_config and make model_arch_config very precise. In model_executor, we use hf_config; In other places in vLLM, we use model_arch_config. This would actually make the migration much easier.

Reply:
IMO this new config should include all things that the engine need and it should be a fixed set of attributes. In model definition (the model executor you mentioned?) we can allow the model vendor to use all of model_arch_config /  hf_config / any other config format that the model vendor likes to use.